content/posts/Jaydip_Dey_Phase_1_Report.md

```diff
@@ -0,0 +1,348 @@
+  - The path for the form is `/questions/new/:question_id`, where `question_id` is a unique string generated using the ShortUniqueID package.
+
+![add-question-nav](/images/Jaydip_GSoC24/add-q-2.webp)
+
+#### Markdown Editor Integration
+
+- **SimpleMDE Markdown Editor:**
+  - The statement input is converted into a markdown editor using [SimpleMDE](https://simplemde.com/)
+
+#### Simulator Integration
+
+- **Circuit Boilerplate and Test Data:**
+  - A hyperlink "Create Circuit BoilerPlate and Test Data" redirects moderators to `/simulator?question_id=:qid` to create and save circuit data.
+
+- **LocalStorage Handling:**
+  - **Storing Progress:**
+    - The question_id and other form values are stored in localStorage when the moderator navigates to the simulator page.
+  - **Loading Progress:**
+    - On the simulator page, if `question_id` is found in localStorage, the corresponding circuit data is loaded:
+
+      ```
+      const urlParams = new URLSearchParams(window.location.search);
+      const questionId = urlParams.get('question_id');
+      if (window.location.pathname.startsWith("/simulator") && questionId && localStorage.getItem(questionId)) {
+          load(JSON.parse(localStorage.getItem(questionId)));
+      }
+      ```
+  
+ ![add-question-simulator](/images/Jaydip_GSoC24/add-q-sim.webp) 
+
+  - **Saving Data:**
+    - When the "Save" button on the simulator page is clicked, the circuit data is saved to localStorage
+
+      ```
+      document.getElementById('saveButton').addEventListener('click', () => {
+          const urlParams = new URLSearchParams(window.location.search);
+          const questionId = urlParams.get('question_id');
+       
+          if (questionId) {
+              let fl = 1;
+              const data = generateSaveData("Untitled", fl);
+              const localStorageKey = `${questionId}`;
+              localStorage.setItem(localStorageKey, data);
+              alert("Circuit boilerplate and test data saved");
+          } else {
+              alert("Error: Something went wrong! Try again");
+          }
+      });
+      ```
+**Add question flowchart**
+![add-question-flowchart](/images/Jaydip_GSoC24/add-q-flow.webp)
+
+#### Dropdowns for Category and Difficulty Level
+
+- **Category:**
+  - Fetched from the database (e.g., logic gates, combinational circuit, sequential circuit).
+- **Difficulty Level:**
+  - Enum values (easy, medium, hard, expert).
+
+**Add question preview**
+{{< video src="/videos/Jaydip_GSoC24/Add-question.mov" type="video/mp4" preload="auto" >}}
+
+For more detailed information, please refer to [this](https://medium.com/@jaydipdey2807/week-5-6-at-circuitverse-google-summer-of-code-2024-87ea6f738cb4) blog post
+
+---
+
+
+### Pull Request
+
+Pull request for phase 1 can be found [here](https://github.com/CircuitVerse/CircuitVerse/pull/5015)
```

Comment:
This PR https://github.com/CircuitVerse/CircuitVerse/pull/4967 should be added in the list also.